### PR TITLE
New Table API

### PR DESCRIPTION
**Pull request by alinastarkov**
Post-merge checklist:

The master branch is auto-deployed to `dash.plot.ly`.
Once you have merged your PR, wait 5-10 minutes and check dash.plot.ly 
to verify that your changes have been made.

- [x] I understand

### CODE REVIEW

File: tutorial/table/styling_chapter.py [hunk 1]
```diff
@@ -413,6 +413,85 @@
...
+        dash_table.DataTable(
+            data=df.to_dict('records'),
+            columns=[{'id': c, 'name': c} for c in df.columns],
+            style_header={ 'border': '1px solid black' },
+            style_cell={ 'border': '1px solid grey' },
+        )
+        '''
+        ),
+
+        dcc.Markdown("## Adding Borders"),
```
Comment (Marc-Andre-Rivet) [inline tutorial/table/styling_chapter.py]: @alinastarkov This supersedes #503, correct?

Comment (Marc-Andre-Rivet) [inline tutorial/table/styling_chapter.py]: @alinastarkov This supersedes #503, correct?

**Comment by Marc-Andre-Rivet** [comment]
This is looking good. Just need 4.0 to be released! Thanks for updating the API update. If this does indeed supersede #503, you can close that one.

File: tutorial/table/styling_chapter.py [hunk 2]
```diff
@@ -243,7 +243,7 @@
                 {"name": ["", "Year"], "id": "year"},
                 {"name": ["City", "Montreal"], "id": "montreal"},
                 {"name": ["City", "Toronto"], "id": "toronto"},
-                {"name": ["City", "Ottawa"], "id": "ottawa", "hidden": True},
+                {"name": ["City", "Ottawa"], "id": "ottawa"},
```
Comment (alexcjohnson) [inline tutorial/table/styling_chapter.py]: AFAICT this is the only knock-on effect of https://github.com/plotly/dash-table/pull/473 in dash-docs, right?

**Pull request merged**